primeqa/qg/models/hybrid_qg/path_sampler.py

```diff
@@ -0,0 +1,197 @@
+import random
+import numpy as np
+
+from sklearn.metrics.pairwise import linear_kernel
+from sklearn.feature_extraction.text import TfidfVectorizer
+
+import stanza
+
+from primeqa.qg.utils.constants import QGSpecialTokens
+
+
+class PathSampler():
```

Comment:
Can you please add some more documentation here and perhaps also add an example of what the chains look like?

Reply:
I have add some more documentation to this file and pushed the changes.